src/csvcubed/utils/validations.py

```diff
@@ -0,0 +1,95 @@
+# This script contain a set of function that can be used to validate specific class attributes/ member variables
+from typing import Optional
+from csvcubed.models.validationerror import ValidateModelProperiesError
+
+from typing import List, TypeVar, Type, Callable
+
+from csvcubed.utils.uri import looks_like_uri
+
+
+T = TypeVar("T")
+
+"""
+This function will valide if the argument provided is in fact a list and,
+in a loop will check each member of the list and return an error if the item in the list is not the correct type. 
```

Comment:
Should say something like: ` and returns any errors returned by the item validation function`